src/components/Comment/index.tsx

```diff
@@ -13,10 +13,42 @@ import {
 } from '../../actions/comments';
 import { LayoutController } from '../../utils/layout';
 import { getNextReplyId } from '../../utils/sequences';
-import CommentReplyComponent from '../CommentReply';
+import CommentReplyComponent, { addRemoteWarningMessage as addRemoteReplyWarningMessage } from '../CommentReply';
+import CommentMessage from '../CommentMessage';
 import type { TranslatableStrings } from '../../main';
 
-async function saveComment(comment: Comment, store: Store) {
+function addRemoteWarningMessage(
+  comment: Comment,
+  message: string,
+  store: Store,
+  addIfLocal = false,
+  onFinish?: () => void
+) {
+  if (addIfLocal || comment.remoteId !== null) {
+    // Only add the warning message if the comment actually exists in the db - editing a comment you've created
+    // locally, for instance, requires no such warning
+    store.dispatch(
+      updateComment(comment.localId, {
+        message: message,
+      })
+    );
+    window.setTimeout(() => {
+      store.dispatch(
+        updateComment(comment.localId, {
+          message: '',
+        })
+      );
+      if (onFinish !== undefined) {
```

Comment:
More idiomatic JS, particularly for a function expression where there is no chance of it having confusing false-y values.

```suggestion
      if (onFinish) {
```

You could also consider rewriting this as a promise, then there would be no need for an extra optional parameter for the callback, and it would be natural for some of this code to have a `.then(` after calling the function, and some to not have it. But that’s a fair amount of extra work.